### PR TITLE
grafana/ui: Move panel-container styles to component

### LINKED ISSUE #49325: Explore: Convert SASS styles of panel-container to emotion
defined in **grafana/public/sass/pages/_dashboard.scss** L10

Affected files in grafana repo:
- [x] public/app/features/explore/Explore.tsx
- [x] public/app/features/explore/NoData.tsx

Other tasks:
- [x] create a component in grafana-ui
- [x] create a related story for Storybook

### PR DESCRIPTION
**Pull request by L-M-K-B**
Fixes #49325

### CODE REVIEW

File: packages/grafana-ui/src/components/PanelContainer/PanelContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { css, cx } from '@emotion/css';
+import React, { DetailedHTMLProps, HTMLAttributes } from 'react';
+
+import { GrafanaTheme2 } from '@grafana/data';
+
+import { useStyles2 } from '../../themes';
+
+type Props = DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>;
+
+const getStyles = (theme: GrafanaTheme2) =>
```
Comment (Clarity-89) [inline packages/grafana-ui/src/components/PanelContainer/PanelContainer.tsx:10]: Nit: Can we have the styles declared after the component, so it's consistent with other components?

File: packages/grafana-ui/src/components/PanelContainer/PanelContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  css`
+    background-color: ${theme.components.panel.background};
+    border: 1px solid ${theme.components.panel.borderColor};
+    border-radius: 3px;
+  `;
+export const PanelContainer = ({ children, className, ...props }: Props) => {
+  const css = useStyles2(getStyles);
```
Comment (Clarity-89) [inline packages/grafana-ui/src/components/PanelContainer/PanelContainer.tsx:17]:
```suggestion
  const styles = useStyles2(getStyles);
```
Nit: Better use `styles` so it doesn't get confused with emotion's `css` function.

File: packages/grafana-ui/src/components/PanelContainer/PanelContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  return (
+    <div className={cx([css, className])} {...props}>
```
Comment (Clarity-89) [inline packages/grafana-ui/src/components/PanelContainer/PanelContainer.tsx:19]:
```suggestion
    <div className={cx(css, className)} {...props}>
```
I don't think array is necessary here?

File: packages/grafana-ui/tsconfig.json [hunk 4]
```diff
@@ -3,7 +3,8 @@
     "baseUrl": "./",
     "declarationDir": "dist",
     "outDir": "compiled",
-    "rootDirs": [".", "stories"]
+    "rootDirs": [".", "stories"],
+    "jsx": "react"
```
Comment (Clarity-89) [inline packages/grafana-ui/tsconfig.json:7]: Wonder why is this necessary?

File: public/app/features/explore/Explore.tsx [hunk 5]
```diff
@@ -368,7 +368,7 @@ export class Explore extends React.PureComponent<Props, ExploreState> {
         {datasourceMissing ? this.renderEmptyState(styles.exploreContainer) : null}
         {datasourceInstance && (
           <div className={cx(styles.exploreContainer)}>
-            <div className={cx('panel-container', styles.queryContainer)}>
+            <PanelContainer className={cx(styles.queryContainer)}>
```
Comment (Clarity-89) [inline public/app/features/explore/Explore.tsx:371]:
```suggestion
            <PanelContainer className={styles.queryContainer}>
```

File: public/app/features/explore/NoData.tsx [hunk 6]
```diff
@@ -2,15 +2,15 @@ import { css, cx } from '@emotion/css';
...
 import { GrafanaTheme2 } from '@grafana/data/src';
-import { useStyles2 } from '@grafana/ui';
+import { useStyles2, PanelContainer } from '@grafana/ui';
 
 export const NoData = () => {
   const css = useStyles2(getStyles);
   return (
     <>
-      <div data-testid="explore-no-data" className={cx([css.wrapper, 'panel-container'])}>
+      <PanelContainer data-testid="explore-no-data" className={cx([css.wrapper])}>
```
Comment (Clarity-89) [inline public/app/features/explore/NoData.tsx:11]:
```suggestion
      <PanelContainer data-testid="explore-no-data" className={css.wrapper}>
```

File: package.json [hunk 7]
```diff
@@ -412,5 +412,10 @@
   "engines": {
     "node": ">= 16"
   },
-  "packageManager": "yarn@3.2.1"
+  "packageManager": "yarn@3.2.1",
+  "dependenciesMeta": {
```
Comment (Clarity-89) [inline package.json:416]: Not familiar with this config option, are we basically pinning eslint version?

**Review by Clarity-89 — COMMENTED** [review]
Thanks for looking into this! Looks like a great start, not sure about some of the changes to package.json and tsconfig. This class name is also used in Enterprise so that will need to be refactored as well before this class name can be completely removed.

File: package.json (hunk 7 above)
[the remark this answers is not in the document]
Comment (L-M-K-B) [reply package.json:416]: **OS**: macOS Monterey 12.4
**IDE**: WebStorm 2022.1.1 

Hi Jack, hi @Clarity-89,
This is the result from enabling ESLint. I first got [error messages](https://raintank-corp.slack.com/archives/CPXKS6FB5/p1653298988675849) regarding ESLint and removed them by following the instructions in the thread. If I delete lines 416-420 I am getting the same errors again.
Do you have any other suggestion how to solve this issue?

File: packages/grafana-ui/tsconfig.json (hunk 4 above)
Comment (L-M-K-B) [reply packages/grafana-ui/tsconfig.json:7]: Line 7 was added to silence this error message:
Screenshot 2022-05-30 at 15 43 47
As this is only a warning and it's not breaking if I remove it, I can also remove it.

File: package.json (hunk 7 above)
[the remark this answers is not in the document]
Comment (Clarity-89) [reply package.json:416]: That seems like an issue with your editor setup, if there's no errors in terminal after you run `yarn start` then it works as expected.

[the remark this answers is not in the document]
Comment (Clarity-89) [reply package.json:416]: If you use Webstorm/Goland you can try these settings: 
Screenshot 2022-05-30 at 16 53 30

Screenshot 2022-05-30 at 16 52 18

[the remark this answers is not in the document]
Comment (L-M-K-B) [reply package.json]: I get why the changes in the package.json are not feasible. The solution above doesn't work so I disabled ESLint for now and will do some more research on the issue.

**Review by Clarity-89 — APPROVED** [review]

File: package.json (hunk 7 above)
[the remark this answers is not in the document]
Comment (Clarity-89) [reply package.json]: @L-M-K-B depending on your version of the editor you can also try this solution: https://youtrack.jetbrains.com/issue/WEB-52594/ESLint-8-0-doesn-t-work-with-YarnPnP#focus=Comments-27-5603740.0-0
